docs/docker.md

```diff
@@ -0,0 +1,45 @@
+# Activate cloud orchestrator in the remote server
+
+This page describes how to run cloud orchestrator managing docker instances
+containing Host Orchestrator inside.
+
+## Prepare docker image
+
+Please follow the docker part of
+[README.md](https://github.com/google/android-cuttlefish/blob/main/README.md#docker)
+in `google/android-cuttlefish` github repository, and check if the docker image
+`cuttlefish-orchestration` exists.
+
+## Build cloud orchestrator
```

Comment:
there's nothing specific when building the binary in the context of docker, it's built as always: `go build ./cmd/cloud_orchestrator`. This step in the documentation as well as the standalone script [scripts/docker/build.sh](https://github.com/google/cloud-android-orchestration/pull/308/commits/c349ce331a5f358b8aec4ad252d8e0b2def630e0#diff-be45b1c53fa8ef7864c013a77325ed29aca3eeae2be39a37e107d63e0d6134d1) makes it seems that we are building for docker in an special way. 

What's is definitely specific to docker is running the binary! In this case, my suggestion is too keep only the `run.sh` script  and make the build command `go build ./cmd/cloud_orchestrator` part of it, then remove the `build.sh` script and the documentation about this step.

Reply:
I moved it from `scripts/docker/build.sh` to `scripts/build_cloud_orchestrator.sh`, not to specify the special way for docker case.

But, I don't want to remove the standalone script, since `go build ./cmd/cloud_orchestrator` itself is relying on current dir. To keep `scripts/docker/run.sh` stable, storing CO binary should be in the known path like root path of git repository, unless users want to specify the path of CO binary. That's the purpose of using standalone script.

Reply:
> But, I don't want to remove the standalone script, since go build ./cmd/cloud_orchestrator itself is relying on current dir.

I think it's not about current dir in this case, building from the same place ensures Consistency and Reproducibility, this principle is followed by tools like `bazel`, `make` and `go build`, where you need to run the build commands where the project is defined. Default `go build` is expected to be run from where `go.mod` is.

We should follow the same principle for the new `run.sh` script as well, where we should fail if we are not where `go.mod` is, keeping it consistent with build system patterns.

Reply:
Understood your saying, and dropped my idea to be independent from current dir. So marking this conversation as resolved in advance.

> We should follow the same principle for the new run.sh script as well, where we should fail if we are not where go.mod is, keeping it consistent with build system patterns.

Just out of curiosity, should we really follow the principle? I think what we should follow is build task running well on where `go.mod` is, not throwing an error if we aren't where `go.mod` is.

Reply:
the first step in run commands, i.e `bazel run` or `go run` is always compiling, then the binary is executed. if we follow the same principle, meaning adding `go build ./cmd/cloud_orchestrator` to `scripts/docker/run.sh` then `run.sh` will fail if not run from where the Golang module is defined. 

you could keep the build step in the documentation, however I suggest to compile the binary as part of `run.sh`.

Reply:
> you could keep the build step in the documentation, however I suggest to compile the binary as part of run.sh, this way the user could skip the build part if wanted.

Sounds good, I haven't considered `go run`. Though, I proceeded to merge this PR in advance to publish the document into the public place. I'll update it in the near future.

Reply:
https://github.com/google/cloud-android-orchestration/pull/311